### PR TITLE
Fix pts with testng

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

Recently we have enabled PTS which causes an error when trying to run testNg tests.
```
Execution failed for task ':core:testNg'.
> Predictive Test Selection only supports JUnit Platform.
```

It has been pointed out that `TestNg` is supported through `junit-platform`.
ref: https://docs.gradle.com/enterprise/predictive-test-selection/#_frameworks_and_languages

I see no reason we shouldn't just use `junit-platform` overall as our default test framework launcher.

ref: https://gradle.slack.com/archives/C05975D6V7H/p1701969512592219

Note that we are unable to add the testNg globally by default due to the following issue: https://github.com/junit-team/testng-engine/issues/87

Modifications:

- Added a test engine `testng` to testing runtime
- Modified to always use `useJUnitPlatform()`

Result:

- We can run testNg tests with PTS enabled

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -127,7 +129,7 @@ allprojects {
         }
 
         predictiveSelection {
-            enabled.set(!isCi)
+            enabled.set(true)
```
[the remark this answers is not in the document]
Comment (jrhee17) [reply build.gradle]: The commit was added to verify that the build passes with pts enabled. I've reverted this commit 🙇

**Review by ikhoon — APPROVED** [review]
Nice changes! 🙇‍♂️🙇‍♂️

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by trustin — APPROVED** [review]
Thanks, @jrhee17! :bow:

**Pull request merged**